File: 832 - Flipping an Image.js
/**
 * @param {number[][]} image
 * @return {number[][]}
 */
var flipAndInvertImage = function(image) {
    let n = image.length;
    for (let i = 0; i < n; i++) {
        let left = 0, right = n - 1;
        while (left < right) {
            if (image[i][left] == image[i][right]) {
                image[i][left] = image[i][right] = image[i][left] ^ 1;
            }
            left++;
            right--;
        }
        if (left == right) {
            image[i][left] ^= 1;
        }
    }
    return image;
};

File: 832 - Flipping an Image.py
class Solution:
    def flipAndInvertImage(self, image: list[list[int]]) -> list[list[int]]:
        n = len(image)
        for i in range(len(image)):
            left = 0
            right = n - 1
            while left < right:
                if image[i][left] == image[i][right]:
                    image[i][left] = image[i][right] = image[i][right] ^ 1
                left += 1
                right -= 1
            if left == right:
                image[i][right] = image[i][right] ^ 1
        return image


    def invert(self, n):
        return 0 if n == 1 else 1